src/Tools/EvtLib/EventLibraryInterface.cxx

```diff
@@ -0,0 +1,309 @@
+//____________________________________________________________________________
+/*
+ Copyright (c) 2003-2020, The GENIE Collaboration
+ For the full text of the license visit http://copyright.genie-mc.org
+
+
+*/
+//____________________________________________________________________________
+
+#include "Framework/Messenger/Messenger.h"
+#include "Framework/Numerical/RandomGen.h"
+#include "Framework/GHEP/GHepRecord.h"
+#include "Framework/GHEP/GHepParticle.h"
+#include "Framework/ParticleData/PDGUtils.h"
+#include "Framework/ParticleData/PDGLibrary.h"
+#include "Framework/ParticleData/PDGCodes.h"
+#include "Framework/Interaction/Interaction.h"
+#include "Tools/EvtLib/EventLibraryInterface.h"
+#include "Tools/EvtLib/EvtLibRecordList.h"
+#include "Tools/EvtLib/Utils.h"
+
+#include "TFile.h"
+
+using namespace genie;
+using namespace genie::evtlib;
+
+//____________________________________________________________________________
+EventLibraryInterface::EventLibraryInterface() :
+  EventRecordVisitorI("genie::evtlib::EventLibraryInterface"),
+  fRecordFile(0)
+{
+
+}
+
+//____________________________________________________________________________
+EventLibraryInterface::EventLibraryInterface(string config) :
+  EventRecordVisitorI("genie::evtlib::EventLibraryInterface", config),
+  fRecordFile(0)
+{
+
+}
+
+//____________________________________________________________________________
+EventLibraryInterface::~EventLibraryInterface()
+{
+  Cleanup();
+}
+
+//____________________________________________________________________________
+void EventLibraryInterface::ProcessEventRecord(GHepRecord * event) const
+{
+// Get event summary constructed by GENIE
+//
+  Interaction* interaction = event->Summary();
+  const InitialState & init_state = interaction->InitState();
+
+  const EvtLibRecord* rec = GetRecord(interaction);
+  if(!rec) return; // Reason has already been printed
+
+  std::unique_ptr<TLorentzVector> probe_p4(init_state.GetProbeP4(kRfLab));
+
+  unsigned int nLep = 0;
+  for(const EvtLibParticle& part: rec->parts) if(pdg::IsLepton(part.pdg)) ++nLep;
+
+  const int firstLep = (nLep == 0) ? -1 : 2;
+  const int lastLep  = (nLep == 0) ? -1 : 1+nLep;
+
+  const int firstHad = (rec->parts.size() == nLep) ? -1 : 2+nLep;
+  const int lastHad  = (rec->parts.size() == nLep) ? -1 : 1+rec->parts.size();
+
+  // Neutrino is a parent to the lepton(s)
+  event->AddParticle(init_state.ProbePdg(),
```

Comment:
If we do this, we destroy the whole point of this exercise because we bypass the flux drivers that specifies the direction